src/libstd/sync/mod.rs

```diff
@@ -42,3 +42,18 @@ mod condvar;
 mod mutex;
 mod once;
 mod rwlock;
+
+/// Save power or switch hyperthreads in a busy-wait spin-loop.  This
+/// function is deliberately more primitive than
+/// `std::thread::yield_now` and does not directly yield to the
+/// system's scheduler.  In some cases it might be useful to use a
+/// combination of both functions.
+#[inline]
+#[stable(feature = "rust1", since = "1.1.6")]
+pub fn spin_loop_pause()
+{
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
```

Comment:
We should probably support more than just x86 if this were to merge - is there some LLVM intrinsic we could leverage here?